invoiceregistry_test.go

```diff
@@ -156,3 +172,56 @@ func TestInvoiceExpiry(t *testing.T) {
 	}
 	cancel3()
 }
+
+func TestAutoSettle(t *testing.T) {
+	c := newRegistryTestContext(t)
+
+	// Subscribe to updates for invoice 1.
+	updateChan, cancelUpdates := c.subscribe(1)
+
+	// Add invoice.
+	c.addInvoice(1, time.Hour, true)
+
+	// Expect an open notification.
+	update := <-updateChan
+	require.Equal(t, persistence.InvoiceStateOpen, update.State)
+
+	preimage := c.preimage(1)
+	resolved := make(chan struct{})
+	c.registry.NotifyExitHopHtlc(&registryHtlc{
+		rHash:         preimage.Hash(),
+		amtPaid:       lnwire.MilliSatoshi(c.testAmt),
+		expiry:        100,
+		currentHeight: 0,
+		resolve: func(r HtlcResolution) {
+			c.logger.Infow("resolution", "res", r)
+			close(resolved)
+		},
+		payload: &testPayload{
+			amt:     lnwire.MilliSatoshi(c.testAmt),
+			payAddr: c.payAddr(1),
+		},
+	})
+
+	update = <-updateChan
```

Comment:
if there's an error and notifications are not sent, the test will wait 10 minutes (by default) until it times out (potentially blocking the CI). Would it be better to create a helper that reads updates from the channel with a much lower timeout?

Something like:

```go
assertFromChan := func(t *testing.T, ch chan<T>, state State) {
  t.Helper()

  select {
    case s := <- ch:
      require.Equal(t, state, s)
    case <- time.After(duration):
      t.Fatalf("timeout")
  }

Reply:
Added `test.Timeout` in a separate commit. This is a technique copied from `lnd`. That we you can still have a short per test timeout without cluttering the test code with selects everywhere.

Reply:
oh that's pretty clever